lib/configure.js

```diff
@@ -272,6 +272,7 @@ function configure (gyp, argv, callback) {
       argv.push('-Dnode_gyp_dir=' + nodeGypDir)
       argv.push('-Dnode_lib_file=' + release.name + '.lib')
       argv.push('-Dmodule_root_dir=' + process.cwd())
+      argv.push('-Dnode_engine=' + (gyp.opts.node_engine || process.jsEngine || 'v8'))
```

Comment:
Sorry for the late reply. The design decision we had taken is other way round. Once user specified `--node-engine`, we will include appropriate headers `deps\v8\include` or `deps\chakrasim\include`. 
E.g. On windows, it is `vcbuild.bat chakracore`
On Linux/Mac , it is `./configure --engine=chakracore`.